src/structures/timedcache.ts

```diff
@@ -0,0 +1,103 @@
+interface ITimedValue<V> {
+    value: V;
+    ts: number;
+}
+
+export class TimedCache<K, V> implements Map<K, V> {
+    private readonly  map: Map<K, ITimedValue<V>>;
+
+    public constructor(private readonly liveFor: number) {
+        this.map = new Map();
+    }
+
+    public clear(): void {
+        this.map.clear();
+    }
+
+    public delete(key: K): boolean {
+        return this.map.delete(key);
+    }
+
+    public forEach(callbackfn: (value: V, key: K, map: Map<K, V>) => void): void {
```

Comment:
why not make this async, in case the callback is async?

Reply:
I'm following the `Map` interface which doesn't let me :(. Though I *guess* I could check to see if callbackfn return a promise and awaiting on that.

Reply:
On second hand, a forEach shouldn't be handling promises. If you want to use promises, use a `for (let a of b) {} ` block

Reply:
Right, true!